core/src/test/scala/org/apache/spark/io/ZStandardBenchmark.scala

```diff
@@ -101,4 +102,28 @@ object ZStandardBenchmark extends BenchmarkBase {
       }
     }
   }
+
+  private def parallelCompressionBenchmark(): Unit = {
+    val N = 128
+    val data = (1 until 128 * 1024 * 1024).toArray
+
+    Seq(3, 9).foreach { level =>
+      val benchmark = new Benchmark(s"Parallel Compression at level $level", N, output = output)
+      Seq(0, 1, 2, 4, 8, 16).foreach { workers =>
```

Comment:
How long does it take?

Reply:
2 ~ 3 secs for case 0 on my mac

Reply:
IIRC, the GitHub Action runner only has 2 CPU cores, and I'm not sure if testing with more workers at this time would yield meaningful test data.

Reply:
```
OpenJDK 64-Bit Server VM 17.0.9+9-LTS on Linux 5.15.0-1051-azure
AMD EPYC 7763 64-Core Processor
Parallel Compression at level 9:          Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parallel Compression with 0 workers                 176            179           2          0.0     1375722.8       1.0X
Parallel Compression with 1 workers                 187            189           4          0.0     1457923.8       0.9X
Parallel Compression with 2 workers                 110            119          13          0.0      858340.6       1.6X
Parallel Compression with 4 workers                 106            109           2          0.0      826473.0       1.7X
Parallel Compression with 8 workers                 110            111           1          0.0      856810.2       1.6X
Parallel Compression with 16 workers                110            113           5          0.0      857306.3       1.6X
```

I have gotten a partial result in GA, it looks fine

Reply:
BTW, according to the result, the best performance is 4-worker? It seems to be a little difficult to find the best fit.

Reply:
In any way, it's good to have this controllability.

Reply:
Based on my observations, we perform best when the value approximately equals the total available cores.

Reply:
The theads max might also be capped by the underlying zstd lib based different OS environments